### PR TITLE
[pulsar-client] Refactor seek to reuse common logic.

### PR DESCRIPTION
**Pull request by sunxiaoguang**
### Motivation
Refactor seek to reuse common logic so adding expression based seek later on would be easier.

### Modifications

*Describe the modifications you've done.*

Change seekAsync(long) and seekAsync(MessageId) in ConsumerImpl to share common code base. 

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is already covered by existing tests

### Does this pull request potentially affect one of the following parts:

NA

### Documentation

NA

### CODE REVIEW

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java [hunk 1]
```diff
@@ -1761,31 +1761,31 @@ public void seek(long timestamp) throws PulsarClientException {
...
+                            + "topic %s to %s", subscription, topicName.toString(), seekBy))));
         }
 
-        final CompletableFuture<Void> seekFuture = new CompletableFuture<>();
+        return Optional.empty();
+    }
 
-        long requestId = client.newRequestId();
-        ByteBuf seek = Commands.newSeek(consumerId, requestId, timestamp);
+    private CompletableFuture<Void> seekAsyncFire(long requestId, ByteBuf seek, String seekBy) {
```
Comment (eolivelli) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java:1781]: what about `seekAsyncInternal` ?

Comment (sunxiaoguang) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java]: Sure, it's fixed.

**Review by eolivelli — APPROVED** [review]
LGTM

**Review by codelipenghui — APPROVED** [review]

**Comment by sunxiaoguang** [comment]
The CI-CPP, Python. Tests / cpp-tests task kept failing for quite some time. Maybe it's something related to docker image upgrade?

**Pull request merged**